implicit/nearest_neighbours.py

```diff
@@ -20,14 +20,14 @@ def fit(self, weighted):
         """ Computes and stores the similarity matrix """
         self.similarity = all_pairs_knn(weighted, self.K).tocsr()
 
-    def recommend(self, userid, user_items, N=10):
+    def recommend(self, userid, user_items, N=10, filter_items=[]):
         """ returns the best N recommendations for a user """
         # calculate the top related items
-        recommendations = user_items[userid].dot(self.similarity)
-        best = sorted(zip(recommendations.indices, recommendations.data), key=lambda x: -x[1])
-
+        recommendations = numpy.squeeze(user_items[userid].dot(self.similarity).toarray())
```

Comment:
I think we should use all the items for the predictions also the ones that have distance = "0". I think this makes algorithmic flow much more predictable.